cGame_logic.cpp

```diff
@@ -1034,177 +987,171 @@ bool cGame::setupGame() {
-
-	// load datafiles
-	gfxdata = load_datafile("data/gfxdata.dat");
-	if (gfxdata == nullptr) {
-		logbook("ERROR: Could not hook/load datafile: gfxdata.dat");
-		return false;
-	} else {
-		logbook("Datafile hooked: gfxdata.dat");
-		memcpy(general_palette, gfxdata[PALETTE_D2TM].dat, sizeof general_palette);
-	}
-
-	gfxaudio = load_datafile("data/gfxaudio.dat");
-	if (gfxaudio == nullptr)  {
-		logbook("ERROR: Could not hook/load datafile: gfxaudio.dat");
-		return false;
-	} else {
-		logbook("Datafile hooked: gfxaudio.dat");
-	}
-
-	gfxinter = load_datafile("data/gfxinter.dat");
-	if (gfxinter == nullptr)  {
-		logbook("ERROR: Could not hook/load datafile: gfxinter.dat");
-		return false;
-	} else {
-		logbook("Datafile hooked: gfxinter.dat");
-	}
-
-	gfxworld = load_datafile("data/gfxworld.dat");
-	if (gfxworld == nullptr) {
-		logbook("ERROR: Could not hook/load datafile: gfxworld.dat");
-		return false;
-	} else {
-		logbook("Datafile hooked: gfxworld.dat");
-	}
-
-	gfxmentat = load_datafile("data/gfxmentat.dat");
-	if (gfxworld == nullptr) {
-		logbook("ERROR: Could not hook/load datafile: gfxmentat.dat");
-		return false;
-	} else {
-		logbook("Datafile hooked: gfxmentat.dat");
-	}
+    /*** Data files ***/
+
+    // load datafiles
+    gfxdata = load_datafile("data/gfxdata.dat");
+    if (gfxdata == nullptr) {
+        logbook("ERROR: Could not hook/load datafile: gfxdata.dat");
+        return false;
+    } else {
+        logbook("Datafile hooked: gfxdata.dat");
+        memcpy(general_palette, gfxdata[PALETTE_D2TM].dat, sizeof general_palette);
+    }
+
+    gfxaudio = load_datafile("data/gfxaudio.dat");
+    if (gfxaudio == nullptr) {
+        logbook("ERROR: Could not hook/load datafile: gfxaudio.dat");
+        return false;
+    } else {
+        logbook("Datafile hooked: gfxaudio.dat");
+    }
+
+    gfxinter = load_datafile("data/gfxinter.dat");
+    if (gfxinter == nullptr) {
+        logbook("ERROR: Could not hook/load datafile: gfxinter.dat");
+        return false;
+    } else {
+        logbook("Datafile hooked: gfxinter.dat");
+    }
+
+    gfxworld = load_datafile("data/gfxworld.dat");
+    if (gfxworld == nullptr) {
+        logbook("ERROR: Could not hook/load datafile: gfxworld.dat");
+        return false;
+    } else {
+        logbook("Datafile hooked: gfxworld.dat");
+    }
+
+    gfxmentat = load_datafile("data/gfxmentat.dat");
+    if (gfxworld == nullptr) {
+        logbook("ERROR: Could not hook/load datafile: gfxmentat.dat");
+        return false;
+    } else {
+        logbook("Datafile hooked: gfxmentat.dat");
+    }
 
     // finally the data repository and drawer interface can be initialized
     m_dataRepository = new cAllegroDataRepository();
     allegroDrawer = new cAllegroDrawer(m_dataRepository);
 
     // randomize timer
-	unsigned int t = (unsigned int) time(0);
-	char seedtxt[80];
-	sprintf(seedtxt, "Seed is %u", t);
-	logbook(seedtxt);
-	srand(t);
+    unsigned int t = (unsigned int) time(0);
```

Comment:
Nit: in modern C++, this is:
```suggestion
    auto t = static_cast<unsigned int>(time(0));
```
Mentions the type only once and on the right side (as usual). There are whole talks about this. :) The style is called: Almost Always Auto.